pennylane/devices/qubit/initialize_state.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions to prepare a state."""
+
+from typing import List
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.interfaces import INTERFACE_MAP
+from pennylane.operation import Operator
+
+INTERFACE_MAP = INTERFACE_MAP.copy()
+del INTERFACE_MAP[None]
+del INTERFACE_MAP["auto"]
+
+
+def _get_padded_wire_order(num_wires, operations):
+    """
+    Gets all distinct wires in ``operations``, and pads that list
+    with new labels if needed.
+
+    Args:
+        num_wires (int): The required number of wires in the returned wire_order
+        operations (List[Operator]): The inputted set of operations to extract wires from
+
+    Returns:
+        wire_order (List[int]): A wire order to use in computations to ensure consistency
+
+    Raises:
+        ValueError: If too many distinct wires are found in the provided list of operations
+    """
+    all_wires = qml.wires.Wires.all_wires([op.wires for op in operations])
```

Comment:
I don't know where it is the case here but if the functions are taken from already existing code, could you mention it on the github comment?